Add unit tests using Jest for the frontend

sonalranjit opened the pull request:
- Setup Jest for frontend project
- Added basic tests for the React components
- Added npm and make targets to run jest tests
- Added Jest test to run on github actions
- Fixed eslint marked violations

closes #26

sonalranjit committed:
working setup of Jest with project

sonalranjit committed:
Add react-leaflet to transformIgnorePatterns

sonalranjit committed:
WIP: writing a first test with Jest

sonalranjit committed:
Mock react for Jest

sonalranjit committed:
working version to test for index page

sonalranjit committed:
added tests for Issues and Location components

sonalranjit committed:
Add tests for Time component

sonalranjit committed:
add tests for Commments components

sonalranjit committed:
add test for Summary components

sonalranjit committed:
remove coverage reports

sonalranjit committed:
add defualt case for switch case in componentToLoad

sonalranjit committed:
update package-lock json

sonalranjit committed:
add make target to run frontend jest test

sonalranjit committed:
run Jest tests on github actions

sonalranjit committed:
fix eslint marked violations

sonalranjit committed:
fix title for github action to deploy landing page

hulloanson reviewed (approved):
The changes are alright. We shall improve codebase to be more test-friendly later.

hulloanson committed:
added @types/jest to shutup IDE

hulloanson committed:
Merge branch 'setup-jest' of https://github.com/bikespace/bikespace into setup-jest

hulloanson committed:
used tsconfig + jest config to correct missing matcher in IDE

- removed import @types/jest as jest test runner does not seem to find them

- added __tests__ to tsconfig.json include

- excluded build from jest.config.json to avoid jest picking up tests as they build

hulloanson commented:
@sonalranjit changed tsconfig and jest.config.json to shut up IDE complaining missing jest matchers. see if it works on your side. jest runs fine on both github workflow and on my machine (node v18). may help to run `npm ci` first in case of conflicting imports / failed test run

sonalranjit commented:
thanks @hulloanson ! 
Yeah i'll create separate issues to refactor the components
